packages/contracts/contracts/generator/GenArt721GeneratorV0.sol

```diff
@@ -0,0 +1,331 @@
+            offset = 65;
+        } else {
+            revert("Unsupported storage version");
+        }
+
+        return AddressChunks.mergeChunks(scriptBytecodeAddresses, offset);
+    }
+
+    /**
+     * @notice Get the project script for a given core contract address and
+     * project ID.
+     * @param coreContractAddress The address of the core contract to retrieve
+     * the project script from.
+     * @param projectId The ID of the project to retrieve the script for.
+     * @return The project script stiched together from all of the script chunks.
+     * @dev This function will revert if the core contract address is not
+     * supported by the DependencyRegistry or if the contract at the address
+     * does not implement the IGenArt721CoreProjectScriptV0 or IGenArt721CoreProjectScriptV1.
+     */
+    function getProjectScript(
+        address coreContractAddress,
+        uint256 projectId
+    ) external view returns (bytes memory) {
+        _onlySupportedCoreContract(coreContractAddress);
+
+        try
+            IGenArt721CoreProjectScriptV1(coreContractAddress)
+                .projectScriptDetails(projectId)
+        returns (string memory, string memory, uint256 scriptCount) {
+            if (scriptCount == 0) {
+                return "";
+            }
+
+            address[] memory scriptBytecodeAddresses = new address[](
+                scriptCount
+            );
+
+            for (uint256 i = 0; i < scriptCount; i++) {
+                scriptBytecodeAddresses[i] = IGenArt721CoreProjectScriptV1(
+                    coreContractAddress
+                ).projectScriptBytecodeAddressByIndex(projectId, i);
+            }
+
+            bytes32 storageVersion = BytecodeStorageReader
+                .getLibraryVersionForBytecode(scriptBytecodeAddresses[0]);
+
+            uint256 offset;
+            if (storageVersion == BytecodeStorageReader.V0_VERSION_STRING) {
+                offset = 104;
+            } else if (
+                storageVersion == BytecodeStorageReader.V1_VERSION_STRING
+            ) {
+                offset = 65;
+            } else {
+                revert("Unsupported storage version");
+            }
+
+            return AddressChunks.mergeChunks(scriptBytecodeAddresses, offset);
+        } catch {
+            // Noop try again for older contracts.
+        }
+
+        try
+            IGenArt721CoreProjectScriptV0(coreContractAddress)
+                .projectScriptInfo(projectId)
+        returns (string memory, uint256 scriptCount) {
+            if (scriptCount == 0) {
+                return "";
+            }
+
+            string memory script;
+            for (uint256 i = 0; i < scriptCount; i++) {
+                string memory scriptChunk = IGenArt721CoreProjectScriptV0(
+                    coreContractAddress
+                ).projectScriptByIndex(projectId, i);
+                script = string.concat(script, scriptChunk);
```

Comment:
I could see this potentially using a lot of gas? which is overall fine for reads of course, but I'm guessing we could improve this with some assembly code in future follow-on activity 👍

Reply:
Yeah, will optimize in a follow on!